packages/ovault-evm/contracts/VaultComposerSyncNative.sol

```diff
@@ -0,0 +1,137 @@
+        SendParam memory _sendParam,
+        address _refundAddress
+    ) external payable nonReentrant {
+        if (msg.value < _assetAmount) revert AmountExceedsMsgValue();
+
+        IWETH(ASSET_ERC20).deposit{ value: _assetAmount }();
+
+        /// @dev Reduce msg.value to the amount used as Fee for the lzSend operation
+        _depositAndSend(
+            OFTComposeMsgCodec.addressToBytes32(msg.sender),
+            _assetAmount,
+            _sendParam,
+            _refundAddress,
+            msg.value - _assetAmount
+        );
+    }
+
+    /**
+     * @dev Unwrap WETH when sending to Stargate PoolNative and send via OFT
+     * @dev Overriden to unwrap WETH when sending to Stargate PoolNative
+     * @param _oft The OFT contract address to use for sending
+     * @param _sendParam The parameters for the send operation
+     * @param _refundAddress Address to receive tokens and native on Pool failure
+     * @param _msgValue The amount of native tokens sent with the transaction
+     */
+    function _sendRemote(
+        address _oft,
+        SendParam memory _sendParam,
+        address _refundAddress,
+        uint256 _msgValue
+    ) internal override {
+        /// @dev _msgValue passed in this call is used as LayerZero fee
+        uint256 msgValue = _msgValue;
+
+        /// @dev Safe because this is the only function in VaultComposerSync that calls oft.send()
+        if (_oft == ASSET_OFT) {
+            /// @dev In deposit's lzReceive() we converted ETH to WETH.
+            /// @dev Incase of redemption the vault outputs WETH.
+            /// @dev So we always have WETH at this point which we unwrap to ETH for the Stargate Pool
+            IWETH(ASSET_ERC20).withdraw(_sendParam.amountLD);
+            /// @dev MsgValue passed to Stargate Pool is nativeFee + amountLD
+            msgValue += _sendParam.amountLD;
+        }
+
+        IOFT(_oft).send{ value: msgValue }(_sendParam, MessagingFee(_msgValue, 0), _refundAddress);
+    }
+
+    /**
+     * @dev Internal function to validate the asset token compatibility
+     * @dev In VaultComposerSyncNative, the asset token is WETH but the OFT token is native (ETH)
+     * @return assetERC20 The address of the asset ERC20 token
+     */
+    function _initializeAssetToken() internal override returns (address assetERC20) {
+        assetERC20 = VAULT.asset();
+
+        if (IOFT(ASSET_OFT).token() != address(0)) revert AssetOFTTokenNotNative();
+
+        /// @dev The asset OFT does NOT need approval since it operates in native ETH.
+        // if (IOFT(ASSET_OFT).approvalRequired()) IERC20(assetERC20).forceApprove(ASSET_OFT, type(uint256).max);
+
+        IERC20(assetERC20).forceApprove(address(VAULT), type(uint256).max);
+    }
+
+    receive() external payable virtual {
+        /// @dev From ASSET_OFT for NativeOFT::lzReceive and from ASSET_ERC20 for WETH::withdraw
+        if (msg.sender != ASSET_OFT && msg.sender != ASSET_ERC20) revert ETHTransferOnlyFromAssetOFT();
```

Comment:
Current error name is not accurate, I'd consider changing it to `ETHTransferNotFromAsset` or `ETHTransferOnlyFromAsset`.